packages/next/client/index.tsx

```diff
@@ -738,14 +735,8 @@ function doRender({
 }
 
 function Root({
-  callback,
   children,
-}: React.PropsWithChildren<{
-  callback: () => void
-}>): React.ReactElement {
-  // We use `useLayoutEffect` to guarantee the callback is executed
-  // as soon as React flushes the update.
-  React.useLayoutEffect(() => callback(), [callback])
```

Comment:
We need `useLayoutEffect` to ensure that e.g. in Concurrent Mode, side effects from the `callback` are run at the right time.

Reply:
I've removed it only because in the callback there was the `onCommit` which should be called before.